app/src/store/playground/playgroundStore.tsx

```diff
@@ -232,6 +248,29 @@ export const createPlaygroundStore = (
         }),
       });
     },
+    setTemplateLanguage: (templateLanguage: TemplateLanguage) => {
+      set({ templateLanguage });
+    },
+    calculateVariables: () => {
+      const instances = get().instances;
+      const variables: Record<string, string> = {};
+      const utils = getTemplateLanguageUtils(get().templateLanguage);
+      instances.forEach((instance) => {
+        // this double nested loop should be okay since we don't expect more than 4 instances
+        // and a handful of messages per instance
+        if (instance.template.__type === "chat") {
+          // for each chat message in the instance
+          instance.template.messages.forEach((message) => {
+            // extract variables from the message content
+            const extractedVariables = utils.extractVariables(message.content);
+            extractedVariables.forEach((variable) => {
+              variables[variable] = "";
+            });
+          });
+        }
+      });
```

Comment:
small thing here since we're basically not supporting completion templates but it would be pretty easy to add an exhaustive switch on the template here and parse variables from the completion templates as well since they are just something like: {content: string } if i remember correctly

Reply:
Done